src/components/BooksForm/BooksForm.jsx

```diff
@@ -1,23 +1,60 @@
+/* eslint-disable import/no-extraneous-dependencies */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import uuid from 'uuid';
+import PropTypes from 'prop-types';
+import { addBook } from '../../actions/index';
+import genID from '../../lib/generateID';
 
 class BookForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      bookCategories: ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'],
+      title: '',
+      category: '',
+      bookCategories: ['', 'Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'],
     };
   }
 
+  handleChange = (event) => {
+    const currentCategory = event.target.value;
+    if (event.target.id === 'category') {
+      this.setState({
+        category: currentCategory,
+      });
+    } else
+    if (event.target.id === 'title') {
+      this.setState({
+        title: currentCategory,
+      });
+    }
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const { title, category } = this.state;
+    const { createBook } = this.props;
+    const book = {
+      id: genID(),
+      title,
+      category,
+    };
+    createBook(book);
+    this.setState({
+      title: '',
+      category: '',
+    });
+  }
+
   render() {
-    const { bookCategories } = this.state;
+    const { bookCategories, category, title } = this.state;
     return (
       <div className="bookForm">
-        <form>
+        <form onSubmit={this.handleSubmit}>
           <label htmlFor="title">Title</label>
```

Comment:
great job for handling `onSubmit`,but where's the submit button, i mean below the form, wont there be a button/something to click after filling form for this submit function to take effect ?? what will invoke this function /?

Reply:
The submit button is at the bottom of the form. This button has type="submit" so React knows that when we click it, it has to call the onSubmit method (declared at the top). So, there are two ways for handling form submission:

1) Using onClick in the button tag
2) Using onSubmit on the from tag

**onSubmit** is more descriptive so we prefer it. React knows how to handle this because the only button with type="submit" in the form will automatically be addressed to the onSubmit method in the form when clicked.